### ISSUE TITLE
Resetting the Ember app breaks with "cannot re-register: serializer:-default".

### ISSUE THREAD

**Issue by kaukas** [emberjs/data#4056]
Hello.

I am working on an Ember + Ember Data based application, writing some tests. Before each test I want to [reset the whole application](https://jsfiddle.net/qmg3surj/1/) but right before the second test this leads to
```
Error: Cannot re-register: `serializer:-default`, as it has already been resolved.
```
What is the recommended way to reset the app? I cannot simply [destroy](https://github.com/emberjs/data/blob/master/tests/helpers/destroy-app.js) and [recreate](https://github.com/emberjs/data/blob/master/tests/helpers/start-app.js) the app as it is done in the Ember Data tests since the app is created once during the page load.

Thank you!

**Comment by fivetanley** [comment]
This does not fix the bug but the recommended way to reset the app is to destroy and recreate the app before each test. What breaks when you do this?

**Comment by kaukas** [comment]
I can not destroy and create it again because the initializing code is part of the single page app which is called only once.

It appears that this is a [known](https://github.com/emberjs/ember.js/issues/12231) [problem](https://github.com/emberjs/ember.js/issues/10310). Strange that I am still getting it with the latest version.

**Comment by fivetanley** [comment]
You should be able to destroy/create and initializers will be called again.

**Comment by bmac** [comment]
@kaukas what version of Ember are you using?

**Comment by kaukas** [comment]
@fivetanley, as I said I can not recreate the app. It's a SPA created once in initializers.

I am on version 2.2.0. I found out that

```js
﻿﻿App.reset();
```
is not enough. Adding


```js
﻿﻿App.buildRegistry();
```
solved the issue for me. It seems strange that `buildRegistry` is not part of `reset` but maybe the authors had their reasons.

In any case this is not an issue of Ember Data in particular. Closing it.
